internal/pkg/skuba/addons/addons.go

```diff
@@ -225,19 +234,26 @@ func (addon Addon) HasToBeApplied(addonConfiguration AddonConfiguration, skubaCo
 		//      now, just return that it doesn't have to be applied.
 		return false, nil
 	}
+	// Check whether this is a CNI addon and whether its base config has been rendered.
+	// If it's a CNI plugin and the base config has not been rendered, we can assume
+	// the user requested a different CNI plugin and this addon does not need to be
+	// applied.
+	if info, _ := os.Stat(addon.addonDir()); addon.AddOnType == CniAddOn && !info.IsDir() {
```

Comment:
Shouldn't you look also at the `error` returned by `os.Stat` and handle that? Especially if `os.IsNotExist(err)`

Reply:
Yep, fixed this.